src/plugins/analysis/information_leaks/code/information_leaks.py

```diff
@@ -63,55 +61,68 @@ class AnalysisPlugin(AnalysisBasePlugin):
             self._find_artifacts(file_object)
-            file_object.processed_analysis[self.NAME]['summary'] = sorted(
-                file_object.processed_analysis[self.NAME])
+            file_object.processed_analysis[self.NAME]['summary'] = sorted(file_object.processed_analysis[self.NAME])
         else:
-            self._find_regex(file_object, file_object.binary, PATH_REGEX)
-            file_object.processed_analysis[self.NAME]['summary'] = sorted(
-                chain(*file_object.processed_analysis[self.NAME].values()))
+            result = _find_regex(file_object.binary, PATH_REGEX)
+            file_object.processed_analysis[self.NAME].update(result)
+            file_object.processed_analysis[self.NAME]['summary'] = sorted(chain(*result.values()))
         return file_object
 
     def _find_artifacts(self, file_object: FileObject):
         for virtual_path_list in file_object.virtual_file_path.values():
             for virtual_path in virtual_path_list:
-                match = self._check_for_files(file_object, virtual_path.split('|')[-1])
-                if match:
-                    continue
-                match = self._check_for_directories(file_object, virtual_path.split('|')[-1])
-                if match:
-                    continue
-                self._find_regex(file_object, virtual_path.split('|')[-1].encode(), FILES_REGEX)
-
-    def _check_for_files(self, file_object: FileObject, file_path: str):
-        for key_path, artifact in PATH_ARTIFACT_DICT.items():
-            if file_path.endswith(key_path):
-                file_object.processed_analysis[self.NAME].setdefault(artifact, []).append(file_path)
-                return True
-        return False
-
-    def _check_for_directories(self, file_object: FileObject, file_path: str):
-        for key_path, artifact in DIRECTORY_DICT.items():
-            file_path_list = file_path.split('/')
-            if len(file_path_list) > 1:
-                if file_path_list[-2] == key_path:
-                    file_object.processed_analysis[self.NAME].setdefault(artifact, []).append(file_path)
-                    return True
-        return False
-
-    def _find_regex(self, file_object: FileObject, search_term, regex_dict):
-        for label, regex in regex_dict.items():
-            result = regex.findall(search_term)
-            if result:
-                result_list = sorted({e.decode(errors='replace') for e in result})
-                file_object.processed_analysis[self.NAME].setdefault(label, []).extend(result_list)
+                file_object.processed_analysis[self.NAME].update(
+                    _check_file_path(get_top_of_virtual_path(virtual_path))
+                )
+
+
+def _check_file_path(file_path: str) -> dict[str, list[str]]:
+    for search_function in (_check_for_files, _check_for_directories, _find_files):
+        results = search_function(file_path)
+        if results:
+            return results
```

Comment:
Is it intendet to ignore the other `search_function`'s once the first one matches?
If it isn't intendet, probably a testcase should be added.

Reply:
Yes, that is the original intention (I only refactored the code to be more "Clean Code compliant")
I added a test for the function to make sure it works as intended